pkg/controller.v1/common/job.go

```diff
@@ -307,9 +307,15 @@ func (jc *JobController) ReconcileJobs(
 				MinResources:      minResources,
 			}
 
-			_, err := jc.SyncPodGroup(metaObject, pgSpec)
+			pg, err := jc.SyncPodGroup(metaObject, pgSpec)
 			if err != nil {
 				log.Warnf("Sync PodGroup %v: %v", jobKey, err)
+				return nil
+			}
+
+			if pg == nil || pg.Status.Phase == "" || pg.Status.Phase == v1beta1.PodGroupPending {
```

Comment:
Could you add some code comments on this, similar to issue description in https://github.com/kubeflow/common/issues/129?

Reply:
done, PTAL